lib/remap-functions/src/parse_glog.rs

```diff
@@ -0,0 +1,190 @@
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            accepts: |v| matches!(v, Value::Bytes(_)),
+            required: true,
+        }]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
+        let value = arguments.required("value")?.boxed();
+
+        Ok(Box::new(ParseGlogFn { value }))
+    }
+}
+
+#[derive(Debug, Clone)]
+struct ParseGlogFn {
+    value: Box<dyn Expression>,
+}
+
+impl Expression for ParseGlogFn {
+    fn execute(&self, state: &mut state::Program, object: &mut dyn Object) -> Result<Value> {
+        let bytes = self.value.execute(state, object)?.try_bytes()?;
+        let message = String::from_utf8_lossy(&bytes);
+
+        let mut log: BTreeMap<String, Value> = BTreeMap::new();
+
+        let captures = REGEX_GLOG
+            .captures(&message)
+            .ok_or("failed parsing glog message")?;
+
+        if let Some(level) = captures.name("level").map(|capture| capture.as_str()) {
+            let level = match level {
+                "I" => Ok("info"),
+                "W" => Ok("warning"),
+                "E" => Ok("error"),
+                "F" => Ok("fatal"),
+                _ => Err(format!(r#"unrecognized log level "{}""#, level)),
+            }?;
+
+            log.insert("level".into(), Value::Bytes(level.to_owned().into()));
+        }
+
+        if let Some(timestamp) = captures.name("timestamp").map(|capture| capture.as_str()) {
+            log.insert(
+                "timestamp".into(),
+                Value::Timestamp(
+                    Utc.datetime_from_str(timestamp, "%Y%m%d %H:%M:%S%.f")
+                        .map_err(|error| {
+                            format!(r#"failed parsing timestamp {}: {}"#, timestamp, error)
+                        })?,
+                ),
+            );
+        }
+
+        if let Some(id) = captures.name("id").map(|capture| capture.as_str()) {
+            log.insert(
+                "id".into(),
+                Value::Integer(id.parse().map_err(|_| "failed parsing id")?),
+            );
+        }
+
+        if let Some(file) = captures.name("file").map(|capture| capture.as_str()) {
+            log.insert("file".into(), Value::Bytes(file.to_owned().into()));
+        }
+
+        if let Some(line) = captures.name("line").map(|capture| capture.as_str()) {
+            log.insert(
+                "line".into(),
+                Value::Integer(line.parse().map_err(|_| "failed parsing line")?),
+            );
+        }
+
+        if let Some(message) = captures.name("message").map(|capture| capture.as_str()) {
+            log.insert("message".into(), Value::Bytes(message.to_owned().into()));
+        }
+
+        Ok(log.into())
+    }
+
+    fn type_def(&self, state: &state::Compiler) -> TypeDef {
+        self.value
+            .type_def(state)
+            .fallible_unless(value::Kind::Bytes)
+            .with_constraint(value::Kind::Map)
```

Comment:
Just a heads up, when #6182 lands, this function will need to define an inner type def to specify the fields that the Map will contain. But that PR isn't there yet, so there's nothing you need to do about it now.